### DISCUSSION TITLE
Need thoughts on a replacment for a figure in the documentation

### DISCUSSION THREAD

**Discussion by IAmMarcelJung** [FPGA-Research/FABulous#209] [General]
For a project report related to FABulous, I created the following figure visualizing the Bitbang protocol using  [tkiz-timing](https://ctan.org/pkg/tikz-timing):

![image](https://github.com/FPGA-Research-Manchester/FABulous/assets/132259776/e7db1209-cc3d-4ef4-be26-8e913c188e15)

In the [simulation setup section](https://fabulous.readthedocs.io/en/latest/simulation/simulation.html) of the documentation, there is this figure:

![image](https://github.com/FPGA-Research-Manchester/FABulous/assets/132259776/f6b6a51e-90e5-4239-bc4f-1505e3a70a6c)

It also describes the Bitbang protocol, but it has two flaws:

1. It does not describe the Bitbang protocol correctly. In the actual Bitbang protocol, the data is transmitted on the rising edge and the control signal is transmitted on the falling edge.
2. It is not vectorized. Even if is perfectly readable, IMHO its always better to have a vectorized image. 

In its original form the image I created is vectorized, but I had to take a screenshot, since it interfered with GitHubs dark theme.

TL;DR: My questions are if my figure would be a good replacement for the existing one and if anyone could suggest some improvements or just drop an opinoin (positive or negative). E.g. I inserted the red lines every time a data bit is captured as `1`. I'm not sure if this is of any use or even hindering. The black vertical lines seperating each 8 bits are useful IMHO.

Looking forward to your comments!

**Comment by KelvinChung2000** [comment]
If you want to make things clearer without further explanation, add a legend to tell what those lines are for. 
I might be misunderstanding the protocol, but you can add an extra row to show what data/bit pattern is stored (in the control register) when data goes high. The best example for demoing is probably writing the 0xFAB0 signal plus a few extra bits.

**Comment by IAmMarcelJung** [comment]
Thanks for your quick comment! `control` and `data` are already the registers where the data is stored. Only `s_clk` and `s_data` are the actually transmitted signals. My example shows the transmission of `0xFAB1`, I should have mentioned that. Maybe it would be good to add a hoizontal line between the acutal data transmission and the bits captured by the Bitbang module. I was also thinking if this was alread too long and a short transfer of 8 eight or so bits would be enough to understand what is happpening. I could add a legend, but still I'm not sure if the image is not too crowded with the red lines. If I replace the figure, I would also add some more explanation about it so it does not necessarily have to stand for itself, although this would of course be nice.

**Reply by KelvinChung2000** [reply]
Showing what the bitbang is capturing will help. I think if you showing what bitbang captured in the register then red line might not be needed. 

It is usually better to let the picture speak for itself since humans generally don't read when they see a picture. Text is extra when they do not fully understand what's going on.

**Reply by IAmMarcelJung** [reply]
Okay then, I think it will be useful to group the singals and add some headlines which tell what the groups are used for. Thanks for your thoughts!

**Reply by IAmMarcelJung** [reply]
![image](https://github.com/user-attachments/assets/4349fcb7-c580-4453-960b-4ada42560937)

Hi, I gave it another try but I think I overcomplicated it for such a simple protocol...
If you have any suggestions to improve it, I'd like to hear them, but it's also okay to just drop it. We should then however still remove the current figure in [the simulation setup section](https://fabulous.readthedocs.io/en/latest/simulation/simulation.html) since it is wrong.

**Reply by KelvinChung2000** [reply]
If the `h` means hex, I think using `h'fab1` might be better since this is how it is normally written in HDL.

**Reply by IAmMarcelJung** [reply]
![image](https://github.com/user-attachments/assets/73220fcc-7ea0-43b2-99ee-0a06866dcb93)


Yes with `h` I meant hex :+1:. Initially I wanted to write `0xXX` but the font size was too tiny then. using `h'xxxx` therefore is a nice suggestion!
